tests/src/Simple/Threading/Threading.cs

```diff
@@ -391,6 +394,25 @@ public static void SemaphoreTest()
     }
 
     [Fact]
+    public static void Reproduce3616()
+    {
+        AutoResetEvent resetEvent = new AutoResetEvent(false);
+        Thread thread = new Thread(() => {
+            var i = 0;
+            var tickCount = Environment.TickCount;
+            while(Environment.TickCount < tickCount + 1000) {
+                i++;
+            }
+            resetEvent.Set();
+            resetEvent.Set();
+        });
+
+        thread.Start();
+        resetEvent.WaitOne(100);
+        thread.Join();
```

Comment:
Should avoid infinite waits in tests, you can use `ThreadTestHelpers.UnexpectedTimeoutMilliseconds` for the timeout